src/main/java/com/projetae/miagiques/exposition/OrganisateurController.java

```diff
@@ -1,58 +1,145 @@
 package com.projetae.miagiques.exposition;
 
 import com.projetae.miagiques.dto.EpreuveDTO;
-import com.projetae.miagiques.entities.Epreuve;
-import com.projetae.miagiques.entities.InfrastructureSportive;
+import com.projetae.miagiques.dto.OrganisateurDTO;
+import com.projetae.miagiques.dto.StatistiqueEpreuveDTO;
+import com.projetae.miagiques.entities.Organisateur;
+import com.projetae.miagiques.entities.Personne;
+import com.projetae.miagiques.metier.EpreuveService;
 import com.projetae.miagiques.metier.OrganisateurService;
+import com.projetae.miagiques.metier.PersonneService;
+import com.projetae.miagiques.metier.StatistiqueEpreuveService;
 import com.projetae.miagiques.utilities.EpreuveExceptions.CapaciteEpreuveSuperieur;
 import com.projetae.miagiques.utilities.EpreuveExceptions.EpreuveExiste;
 import com.projetae.miagiques.utilities.EpreuveExceptions.EpreuveInexistante;
-import com.projetae.miagiques.utilities.InfrastructureSportiveExceptions.InfrastructureSportiveInexistant;
+import com.projetae.miagiques.utilities.InfrastructureSportiveExceptions.InfrastructureSportiveInexistante;
+import com.projetae.miagiques.utilities.PersonneExceptions.CompteExiste;
+import com.projetae.miagiques.utilities.PersonneExceptions.CompteInexistant;
+import com.projetae.miagiques.utilities.PersonneExceptions.RoleIncorrect;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-
-import java.sql.Date;
-import java.sql.Timestamp;
 import java.text.ParseException;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.Collection;
+
 
 @RestController
-@RequestMapping("/organisateurs")
+@RequestMapping("/organisateurs/{emailUtilisateur}")
 
 public class OrganisateurController {
 
     @Autowired
-    private OrganisateurService organisateurService ;
+    private EpreuveService epreuveService ;
+
+    @Autowired
+    private StatistiqueEpreuveService statistiqueEpreuveService ;
+
+    @Autowired
+    private PersonneService personneService ;
+
+    @Autowired OrganisateurService organisateurService ;
+    /**
+     *
+     * @param mail représente le mail de la personne qui essaye de faire une action
+     * @return true si l'action est valide avec le compte mentionné
+     * @throws CompteInexistant si le compte avec l'adresse email mentionné n'existe pas
+     * @throws RoleIncorrect si le role ne correspond pas a la tache qu'on essaye de faire
+     */
+    public boolean testerRole(String mail) throws CompteInexistant, RoleIncorrect {
+        ArrayList<Personne> listeRolesPersonne = new ArrayList<>(this.personneService.seConnecter(mail)) ;
+
+        int i = 0;
+        do {
```

Comment:
alternative pour éviter un do while et if imbriqués et les variables en plus :

Collection<Personne> listeRolesPersonne = new Collection<>(this.personneService.seConnecter(mail));

if (!listeRolesPersonne.stream().anyMatch( p -> p instanceof Organisateur)){
    throw new RoleIncorrect();
}

return true;